airflow/auth/managers/simple/simple_auth_manager.py

```diff
@@ -80,6 +81,9 @@ class SimpleAuthManager(BaseAuthManager[SimpleAuthManagerUser]):
     # Cache containing the password associated to a username
     passwords: dict[str, str] = {}
 
+    # TODO: Needs to be deleted when Airflow 2 legacy UI is gone
+    appbuilder: AirflowAppBuilder | None = None
```

Comment:
Hmm, why does simple auth manager need this? If it's just for the assignment, can we check before we set it? Or is this for typing/linting? Having this in every auth manager, even those that have nothing to do with FAB, doesn't feel very good imo.

Reply:
I agree but at the same time the auth managers, today, need FAB and I actually like having it declare in all the auth managers so everyone is aware. All auth managers need to create custom views for their own use. Example: the simple auth manager define its own login form. It adds this view to the environment using FAB. Today we do not have mechanism to add views to Airflow 3. I was counting on AIP-68 but it seems it is kind of blocked.

I added this one initially for typing and I am pretty sure we could remove it and work it out but again, I kinda like it is there so that we know that all auth managers need FAB.

Reply:
> Today we do not have mechanism to add views to Airflow 3. I was counting on AIP-68 but it seems it is kind of blocked.

I mean, that's what this PR is doing :)

I'm okay if we keep it short term so we can do changes in stages, but before we actually release AF3 I think it should go. Simple auth manager will already have to refactor this stuff - there will be no `AirflowBaseView` in core.

Reply:
Oh yes, 100% agree :) This is short term and it will go away before AF3 release. Sorry I did not mention it ... As you mention, I rather do it in stages (the PR is already big enough)

Reply:
> I mean, that's what this PR is doing :)

I actually thought this PR was only to extend the Airflow API but we could also use it for views/pages I guess

Reply:
It's just an app, it can do anything :)

Reply:
I just realized that :)